sys/netpfil/pf/if_pfsync.c

```diff
@@ -1384,19 +1511,20 @@ pfsyncioctl(struct ifnet *ifp, u_long cmd, caddr_t data)
 			sifp = NULL;
 		else if ((sifp = ifunit_ref(pfsyncr.pfsyncr_syncdev)) == NULL)
 			return (EINVAL);
-
-		if (sifp != NULL && (
-		    pfsyncr.pfsyncr_syncpeer.s_addr == 0 ||
-		    pfsyncr.pfsyncr_syncpeer.s_addr ==
-		    htonl(INADDR_PFSYNC_GROUP)))
+		if (sifp != NULL && (sc->sc_sync_peer.sa.sa_family == AF_UNSPEC)) {
 			imf = ip_mfilter_alloc(M_WAITOK, 0, 0);
+		}
 
 		PFSYNC_LOCK(sc);
-		if (pfsyncr.pfsyncr_syncpeer.s_addr == 0)
-			sc->sc_sync_peer.s_addr = htonl(INADDR_PFSYNC_GROUP);
-		else
-			sc->sc_sync_peer.s_addr =
-			    pfsyncr.pfsyncr_syncpeer.s_addr;
+		printf("pfsync: Address family: %d\n", sc->sc_sync_peer.sa.sa_family);
+		if (sifp != NULL && (sc->sc_sync_peer.sa.sa_family == AF_UNSPEC)) {
```

Comment:
It's the same condition as above. You could have  just one `if` and merge those 2 code blocks. Just watch out for `PFSYNC_LOCK`.